src/pages/groupComparison/Survival.tsx

```diff
@@ -389,19 +390,27 @@ export default class Survival extends React.Component<ISurvivalProps, {}> {
 
                                 for (const s of patientSurvivals[prefix]) {
                                     // count the number of patients in each active group
-                                    const groups =
+                                    const groupUids =
                                         patientToAnalysisGroups[
                                             s.uniquePatientKey
                                         ] || [];
-                                    for (const groupName of groups) {
-                                        numPatientsPerGroup[groupName] += 1;
+                                    for (const uid of groupUids) {
+                                        numPatientsPerGroup[
+                                            uidToAnalysisGroup[uid].name
+                                        ] += 1;
                                     }
                                 }
                                 return {
                                     prefix,
                                     displayText,
-                                    numPatients:
-                                        patientSurvivals[prefix].length,
+                                    numPatients: _.sumBy(
+                                        patientSurvivals[prefix],
+                                        s =>
+                                            +(
```

Comment:
what does that plus operator do?

Reply:
seems weird adding a boolean?

Reply:
+ converts true to 1 and false to 0

Reply:
I feel like we should avoid this kind of wizardry/coercion in favor of + (s ? 1 : 0)